src/table/use-sticky-columns.tsx

```diff
@@ -0,0 +1,229 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useLayoutEffect, useState, createRef, useEffect, useCallback, useMemo } from 'react';
+import { TableProps } from './interfaces';
+import clsx from 'clsx';
+interface CellOffsets {
```

Comment:
The same interface is declared in `table/internal.tsx` Would it make sense to de-duplicate it? I'd move it to `interfaces.ts`